<launch>
    <!-- 启动摄像头节点 -->
    <group ns="camera">
        <node pkg="libuvc_camera" type="camera_node" name="mycam">
        <!-- Parameters used to find the camera -->
        <param name="vendor" value="0x09da"/>
        <param name="product" value="0x2700"/>
        <param name="serial" value=""/>
        <!-- If the above parameters aren't unique, choose the first match: -->
        <param name="index" value="0"/>

        <!-- Image size and type -->
        <param name="width" value="1280"/>
        <param name="height" value="720"/>
        <!-- choose whichever uncompressed format the camera supports: -->
        <param name="video_mode" value="mjpeg"/> <!-- or yuyv/nv12/mjpeg -->
        <param name="frame_rate" value="30"/>

        <param name="timestamp_method" value="start"/> <!-- start of frame -->
        <!-- <param name="camera_info_url" value="file:///tmp/cam.yaml"/> -->

        <param name="auto_exposure" value="3"/> <!-- use aperture_priority auto exposure -->
        <param name="auto_white_balance" value="false"/>
        </node>
    </group>
    <!-- OpenCV节点 -->
    <node name="open_cv_gray" 
          pkg="open_cv" 
          type="open_cv_gray" 
          output="screen"
          respawn="false">
        <!-- 可通过参数服务器设置订阅和发布的topic -->
        <param name="input_topic"  value="/camera/image_raw"/>
        <param name="output_topic" value="/image_converter/gray"/>
    </node>
</launch>